File: ios/Runner/NativeVideoViewFactory.swift
import Flutter
import UIKit

class NativeVideoViewFactory: NSObject, FlutterPlatformViewFactory {
    private var channel: FlutterMethodChannel!
    init(messenger: FlutterBinaryMessenger) {
        super.init()
        channel = FlutterMethodChannel(name: "native_video/channel", binaryMessenger: messenger)
        channel.setMethodCallHandler(handle)
        NativeVideoEventBus.channel = channel
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol { return FlutterStandardMessageCodec.sharedInstance() }

    func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        let dict = args as? [String: Any]
        let logicalId = (dict?["id"] as? Int) ?? Int(viewId)
        let view = NativeVideoView(frame: frame)
        view.logicalId = logicalId
        NativeVideoRegistry.views[logicalId] = view
        return view
    }

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any], let id = args["id"] as? Int, let view = NativeVideoRegistry.views[id] else {
            result(FlutterError(code: "VIEW_NOT_FOUND", message: "View not found", details: nil)); return
        }
        switch call.method {
        case "register":
            let url = args["url"] as? String ?? ""
            let autoPlay = args["autoPlay"] as? Bool ?? false
            let loop = args["loop"] as? Bool ?? false
            let volume = args["volume"] as? Double ?? 1.0
            let muted = args["muted"] as? Bool ?? false
            view.prepare(url: url, autoPlay: autoPlay, loop: loop, volume: Float(volume), muted: muted)
            result(nil)
        case "play":
            view.play(); result(nil)
        case "pause":
            view.pause(); result(nil)
        case "seekTo":
            let ms = args["ms"] as? Int ?? 0
            view.seekTo(milliseconds: ms); result(nil)
        case "setVolume":
            let volume = args["volume"] as? Double ?? 1.0
            view.setVolume(Float(volume)); result(nil)
        case "setMuted":
            let muted = args["muted"] as? Bool ?? false
            view.setMuted(muted); result(nil)
        case "dispose":
            view.dispose()
            NativeVideoRegistry.views.removeValue(forKey: id)
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/app/src/main/kotlin/com/example/native_video_player/NativeVideoViewFactory.kt
package com.example.native_video_player

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeVideoViewFactory(private val messenger: BinaryMessenger) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE),
    MethodChannel.MethodCallHandler {

    private val channel = MethodChannel(messenger, "native_video/channel")

    init {
        channel.setMethodCallHandler(this)
        NativeVideoEvents.channel = channel
    }

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val params = args as? Map<*, *>
        val logicalId = (params?.get("id") as? Int) ?: viewId
        val view = NativeVideoView(context)
        view.logicalId = logicalId
        NativeVideoRegistry.put(logicalId, view)
        return view
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        val id = (call.argument<Int>("id"))
        val view = id?.let { NativeVideoRegistry.get(it) }
        when (call.method) {
            "register" -> {
                val url = call.argument<String>("url")
                val autoPlay = call.argument<Boolean>("autoPlay") ?: false
                val loop = call.argument<Boolean>("loop") ?: false
                val volume = call.argument<Double>("volume") ?: 1.0
                val muted = call.argument<Boolean>("muted") ?: false
                if (view != null && url != null) {
                    view.prepare(url, autoPlay, loop, volume.toFloat(), muted)
                    result.success(null)
                } else result.error("VIEW_OR_URL_NULL", "View or url is null", null)
            }
            "play" -> { view?.play(); result.success(null) }
            "pause" -> { view?.pause(); result.success(null) }
            "seekTo" -> {
                val ms = call.argument<Int>("ms")?.toLong() ?: 0L
                view?.seekTo(ms); result.success(null)
            }
            "setVolume" -> {
                val vol = call.argument<Double>("volume")?.toFloat() ?: 1f
                view?.setVolume(vol); result.success(null)
            }
            "setMuted" -> {
                val muted = call.argument<Boolean>("muted") ?: false
                view?.setMuted(muted); result.success(null)
            }
            "dispose" -> {
                view?.dispose()
                if (id != null) NativeVideoRegistry.remove(id)
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }
}
